tests/transformer/test_iterator.py

```diff
@@ -0,0 +1,171 @@
+from RestrictedPython.Guards import guarded_iter_unpack_sequence
+from tests import e_exec
+
+import pytest
+import RestrictedPython
+import types
+
+
+ITERATORS = """
+def for_loop(it):
+    c = 0
+    for a in it:
+        c = c + a
+    return c
+
+
+def nested_for_loop(it1, it2):
+    c = 0
+    for a in it1:
+        for b in it2:
+            c = c + a + b
+    return c
+
+def dict_comp(it):
+    return {a: a + a for a in it}
+
+def list_comp(it):
+    return [a + a for a in it]
+
+def nested_list_comp(it1, it2):
+    return [a + b for a in it1 if a > 1 for b in it2]
+
+def set_comp(it):
+    return {a + a for a in it}
+
+def generator(it):
+    return (a + a for a in it)
+
+def nested_generator(it1, it2):
+    return (a+b for a in it1 if a > 0 for b in it2)
+"""
+
+
+@pytest.mark.parametrize(*e_exec)
+def test_RestrictingNodeTransformer__guard_iter(e_exec, mocker):
```

Comment:
As there is another test function in this module for `guard_iter` its name should end in `__1`.